src/network/networkd-dhcp4.c

```diff
@@ -1319,9 +1319,30 @@ int dhcp4_configure(Link *link) {
         if (r < 0)
                 return log_link_warning_errno(link, r, "DHCP4 CLIENT: Failed to set callback: %m");
 
-        r = sd_dhcp_client_set_request_broadcast(link->dhcp_client, link->network->dhcp_broadcast);
-        if (r < 0)
-                return log_link_warning_errno(link, r, "DHCP4 CLIENT: Failed to set request flag for broadcast: %m");
+        /* Some interfaces require that the DHCPOFFER message is being broadcast because
+         * they can't handle unicast messages while not fully configured.
+         * Setting device property ID_NET_NEEDS_DHCP_BROADCAST to "1" can be used to
+         * override the dhcp_broadcast flag for such interfaces. */
+        dhcp_broadcast = link->network->dhcp_broadcast;
+        if (dhcp_broadcast < 0 && link->sd_device) {
+                const char *val;
+
+                if (sd_device_get_property_value(link->sd_device, "ID_NET_NEEDS_DHCP_BROADCAST", &val) >= 0) {
+                        r = parse_boolean(val);
+                        if (r < 0)
+                                log_link_warning_errno(link, r, "DHCP4 CLIENT: Failed to parse ID_NET_NEEDS_DHCP_BROADCAST, ignoring: %m");
+                        else {
+                                dhcp_broadcast = r;
+                                log_link_info(link, "DHCP4 CLIENT: Overriding broadcast flag.");
```

Comment:
Have you read my previous suggestion? This block does not _override_ `RequestBroadcast=` setting. Please adjust the log message. Also, I think it is not necessary to log in LOG_INFO level. Please log in debug level.

Reply:
By overriding, I meant overriding the default, but obviously failed to convey that. Would "Overriding the default setting of the broadcast flag" be adequate?

Reply:
I think "override" is ambiguous anyway here, as `sd_dhcp_client::request_broadcast` and `RequestBroadcast=` are `false` by default, and `ID_NET_NEEDS_DHCP_BROADCAST` property may be zero, in that case the default is not overridden.
So, without "override", how about the following:
```
log_link_debug(link, "DHCP4 CLIENT: %s broadcast flag based on ID_NET_NEEDS_DHCP_BROADCAST udev property.", dhcp_broadcast ? "enabling" : "disabling");
```

Reply:
Makes sense, will update accordingly.